### DISCUSSION TITLE
how is use 'use client'?

### DISCUSSION THREAD

**Discussion by duoluodexiaoxiaoyuan** [vercel/next.js#55194] [Help] [answered]
### Summary

for example
![image](https://github.com/vercel/next.js/assets/59086082/0b59016b-2176-471f-8b33-9795733607f0)
![image](https://github.com/vercel/next.js/assets/59086082/7a685942-7baa-4a5e-b4c7-cbe62292e794)

interface is ok.

But

![image](https://github.com/vercel/next.js/assets/59086082/b8148366-8715-4655-86e7-a651786e0507)

my vscode terminal and website console  both  print  message,  more important   interface is call four times, I don't reason.

![image](https://github.com/vercel/next.js/assets/59086082/c02af403-32fa-4580-8c4a-701ed38669aa)
![image](https://github.com/vercel/next.js/assets/59086082/125a64af-dcab-4717-8352-6627d9d0246d)

### Additional information

```js
this is my github repo: https://github.com/duoluodexiaoxiaoyuan/mynextproject/tree/useRequestWithNext
```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

`use client` creates a Client Side Tree. Therefore, every child of the Client Side Tree becomes a client component itself.

This means that `EveryWord` is being made into a client component by the page parent, which you are opting into `use client`.

There's ways to weave together Client Components inside a Server Component Tree, by using "children"-props, https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props, where you use:

- a server component that does something you want
- a client component that does interactivity you want
- a server component that puts together the other two

I'd recommend reading:

- https://github.com/reactwg/server-components/discussions/4

**Comment by duoluodexiaoxiaoyuan** [comment]
> Hi,
> 
> `use client` creates a Client Side Tree. Therefore, every child of the Client Side Tree becomes a client component itself.
> 
> This means that `EveryWord` is being made into a client component by the page parent, which you are opting into `use client`.
> 
> There's ways to weave together Client Components inside a Server Component Tree, by using "children"-props, https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props, where you use:
> 
> * a server component that does something you want
> * a client component that does interactivity you want
> * a server component that puts together the other two
> 
> I'd recommend reading:
> 
> * [Why do Client Components get SSR'd to HTML? reactwg/server-components#4](https://github.com/reactwg/server-components/discussions/4)

thank u replay
